### ISSUE TITLE
[ENH] Cell type identification

### ISSUE THREAD

**Issue by mstrazar** [biolab/orange3-single-cell#337]
Implement a widget that annotates a 2D projection of cell types.
It operates in multiple modular steps:

1. Assign cell type significance scores to each cell. This is computed only once and must be efficient. This module will be implemented in https://github.com/biolab/orange3-bioinformatics/pull/133
2. Compute a 2D projection (default: t-SNE, use optional Preprocessor). 
3. Find regions on the 2D projection (default: DBscan, optional: use clustering variable) that are enriched with certain cell types. Allow modifying resolution. This will be implemented in https://github.com/biolab/orange3-bioinformatics/pull/134. Every group will be circled with the concave hull. This functionality will be also implemented in https://github.com/biolab/orange3-bioinformatics/pull/134.
4. Visualize regions with the clearest enrichment of cell types. 

#### Workflow

![image1](https://user-images.githubusercontent.com/5073833/57526516-dc677100-732d-11e9-8b33-2fccef2431c8.png)

#### Widget

The widget will be similar to the scatter plot. It will include the control area and the visualization area.

##### Control area

It includes three main sections:

- Infobox at the top
- Annotation box with widget specific settings
    - `epsilon` - the setting which will be passed to DBSCAN clustering. The decision is that the user can set this value manually. And reset manual selection with a reset button. After the selection is reset the button is deactivated. When a user does not provide its own setting for epsilon, it is set using a specific formula which will be implemented in https://github.com/biolab/orange3-bioinformatics/pull/134.
    - `p-value threshold` - the threshold value which is passed to the annotation module. The value is between 0 and 1.
    - `show hull` - checkbox which shows and hides the hull around clusters. Default: True
    - `show labels` - checkbox which shows and hides labels for the cell type from the graph. Default: True
    - `num of labels` - number of labels shown per cluster. Default: 1. Allow up to three. 
- Other scatter plot specific settings.

##### Visualization area

This part is a scatter plot like a graph with additional elements:

- Each cluster is enclosed with a hull.
- Each cluster has annotations somewhere in the middle of the cluster. The number of annotations per cluster is specified in the control area.
- Each cluster has a tooltip (on mouse over) which shows all annotations and their scores.

Examples of good visualizations:

![screen2](https://user-images.githubusercontent.com/5073833/57526894-ea69c180-732e-11e9-85e3-4c3b60bfe597.png)
![screen3](https://user-images.githubusercontent.com/5073833/57526900-edfd4880-732e-11e9-9e48-79a7705f18eb.png)

**Inputs:**

- Data (gene expression matrix)
- Markers (table linking genes to cell types, the output of Marker genes)
- Preprocessor (optional; t-SNE, MDS, PCA, ...)

**Outputs:**

- Data (gene expression matrix augmented with **cell type scores** and **projection coordinates** as meta values)
- Selected Data (selected gene expression matrix augmented with **cell type scores** and **projection coordinates** as meta values)

**Comment by BlazZupan** [comment]
This issue is indeed single-cell oriented but could be implemented in Orange Bioinformatics add-on since we could use the same functionality for, say, bulk expression analysis. We could split it into several smaller issues first. I propose this issue is kept open and closed once we publish the first blog describing the results on scOrange web site.

**Comment by PrimozGodec** [comment]
Closed via https://github.com/biolab/orange3-bioinformatics/issues/135
